packages/config/index.js

```diff
@@ -81,24 +82,29 @@ function applyEnvironmentConfig(config) {
 }
 
 function resolvePlaceholders(config) {
-  var keys = Object.keys(config);
-  var regExp = new RegExp('<(' + keys.join('|') + ')>', 'g');
+  var regExp = new RegExp('<(' + Object.keys(config).join('|') + ')>', 'g');
   function replaceRecursive(item) {
     if (Array.isArray(item)) {
       return item.map(replaceRecursive);
     }
+    if (isPlainObject(item)) {
+      return Object.keys(item).reduce(function(result, key) {
+        result[key] = replaceRecursive(item[key]);
+        return result;
+      }, {});
+    }
     if (typeof item === 'string') {
-      return item.replace(regExp, function(_, match) {
-        var result = config[match].toString();
+      return item.replace(regExp, function(_, configKey) {
+        var result = config[configKey];
+        if (typeof result !== 'string') {
+          result = String(result).toString();
```

Comment:
What is the reason for `String(x).toString()` instead of just `String(x)`?

Reply:
I just wanted to make it obvious that we are dealing with a primitive string here - I always found it a bit confusing that..:

```js
'foo' === 'foo'
true
String('foo') === String('foo')
// true
String('foo') === new String('foo')
// false
```

But maybe that's just me..?

Reply:
(of course I have come to terms with what is going on in there, but anyway...)